nxcomp/Log.cpp

```diff
@@ -0,0 +1,81 @@
+
+#include <stdio.h>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+
+#include "Log.h"
+
+NXLog nx_log;
+
+
+NXLog::NXLog(char *filename)
+{
+	std::fstream *file = new std::fstream();
+	file->open(filename);
+	stream_ = file;
+	level_ = DEBUG;
+}
+
+bool NXLog::will_log() const
+{
+	std::map<std::string, NXLogLevel>::const_iterator item = per_file_levels_.find(current_file());
+
+	if ( item != per_file_levels_.end() )
+	{
+		return current_level_ >= item->second;
+	}
+	else
+	{
+		return current_level_ >= level_;
+	}
+}
+
+
+std::string NXLogStamp::to_string() const
+{
+	static const char* level_names[] = {
+		"DBG",
+		"INF",
+		"WRN",
+		"ERR",
+		"FAT"
```

Comment:
That was to save space in the log output, to have more room for useful information. But sure, I can change it.